kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -112,7 +112,7 @@ impl<'tcx> GotocCtx<'tcx> {
         place: &Place,
         span: Span,
     ) -> Stmt {
-        let intrinsic_sym = instance.mangled_name();
+        let intrinsic_sym = instance.trimmed_name();
```

Comment:
@celinval what is the difference between `mangled_name` and `trimmed_name` in StableMIR?

Reply:
`mangled_name` is the unique name of this specific instance which gets codegen. `trimmed_name` is the Rust name in a given context. It differs from `name()` as it tries to trim the modules whenever there's no ambiguity.